browser/src/control/jsdialog/Widget.MenuButton.js

```diff
@@ -130,8 +134,16 @@ function _menubuttonControl (parentContainer, data, builder) {
 				JSDialog.AddOnClick(control.label, clickFunction);
 		}
 
-		if (control.arrow)
+		if (control.arrow) {
 			JSDialog.AddOnClick(control.arrow, clickFunction);
+			builder.map.on('jsdialog', function (e) {
```

Comment:
This code is duplicated, also best to not put many places for processing the message queue as it might later cause hard to find bugs.

It goes: `socket -> JSDialogMessageRouter -> JSDialog component` and only the "component" should execute any operations, calling some widget actions if needed.

Possible solution:
- please put in the widget's root (`control.container`) function: `onDropdownClose`
- Dropdown widget has already information about widget from which it was created (see `JSDialog.OpenDropdown(dropdownId, control.container <---`)
- Dropdown when closing, can call the `onDropdownClose` if exists

https://github.com/CollaboraOnline/online/blob/836aa22dad3d1d64668e213e9749f71b3ffedc14/browser/src/control/jsdialog/Util.Dropdown.js

Reply:
Interesting your suggestion, I have traced events when dropdown is closed, which exists JSDialog.CloseDropdown and 'clickfunction' callbacks, I did not want to do invasive changes. Well let's try it.